source/extensions/filters/http/transformation/inja_transformer.cc

```diff
@@ -207,6 +276,15 @@ void InjaTransformer::transform(Http::HeaderMap &header_map,
     replace_body(output);
   }
 
+  // DynamicMetadata transform:
+  for (const auto &templated_dynamic_metadata : dynamic_metadata_) {
```

Comment:
Potentially most work regarding dynamic metadata could be done in the log callback because we are not actually planning on using it for any other purpose than logging

Reply:
right, but we'll potentially need access to the body, hence why it needs to be here